test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -28,6 +28,41 @@ Describe "TabCompletion" -Tags CI {
         $res.CompletionMatches[0].CompletionText | Should -BeExactly 'notepad.exe'
     }
 
+    It 'Should not include duplicate command results' {
+        $OldModulePath = $env:PSModulePath
+        $tempDir = Join-Path -Path $TestDrive -ChildPath "TempPsModuleDir"
+        try
+        {
+            $ModuleDirs = @(
+                Join-Path $tempDir "TestModule1\1.0"
+                Join-Path $tempDir "TestModule1\1.1"
+                Join-Path $tempDir "TestModule2\1.0"
```

Comment:
1. I agree it is useful to remove duplicates for the _same_ modules with different versions.
2. I'd prefer to get variants with full names (`TestModule1\MyTestFunction` and `TestModule2\MyTestFunction`) for different modules.
3. In original issue there are two cases - here I see test for one case only.

Reply:
1: Glad we agree on that.
2: That is exactly what I'm doing:
```
$SortedMatches[0] | Should -Be "TestModule1\MyTestFunction"
$SortedMatches[1] | Should -Be "TestModule2\MyTestFunction"
```

The logic works like this: If a command name is available in 2 different modules, neither of which have been imported, then it shows the module qualified name for both.
If one of the modules have been imported then it will show the short name for that one, and the module qualified name for the other.  
If both modules have been imported then it will show both of them with the module qualified name.

3: Good point, I will look into that later.

Reply:
I've looked more into it. I don't think there's a need for a separate case for scenario 2 because the code handles duplicates no matter how it gets them. So it doesn't matter if it's due to the nested modules issue, multiple versions, or separate modules. It handles them all the same way.